sorting/countingSort.c

```diff
@@ -1,28 +1,45 @@
 #include <stdio.h>
-  int main()
- {
- 	int i,n,l=0;
-  scanf("%d",&n);
- 	int a[n];
-  for(i=0;i<n;i++)
- 	{
-  	 	scanf("%d",&a[i]);
-  	 	if(a[i]>l)
-  	 	 l=a[i];
- 	}
- 	int b[l+1]={0};
- 	for(i=0;i<n;i++)
- 	 b[a[i]]++; //hashing number to array index
- 	for(i=0;i<(l+1);i++)
- 	{
- 		if(b[i]>0)
- 	{
- 			while(b[i]!=0) //for case when number exists more than once
- 			{
- 			printf("%d ",i); 
- 			b[i]--;
- 			}
-    	}
- 	}
- 	return 0;
- } 
\ No newline at end of file
+#include <string.h>	// for memset
+#include <assert.h>
+
+int main()
+{
+	int i, n, l = 0;
+	scanf("%d", &n);
+
+	assert(n < 1e7);
```

Comment:
Why do you use a magic number? Simply use constant number for initial array

Reply:
The maximum length of array is fixed due to memory constraint of mordern day computer ( array occupies contigous memory block). So it is advisable not to try to create a huge sized array. The max size of locally defined array is roughly 1e7 ie 10^7 . [reference](https://discuss.codechef.com/t/maximum-size-of-an-array/3633)

Reply:
It just illustrates how counting sort works, so using constant number  (ex: 256) is good enough.

Reply:
@danghai Fine. Updated accordingly.